app/scripts/controllers/transactions/index.js

```diff
@@ -606,12 +607,23 @@ export default class TransactionController extends EventEmitter {
 
         const metametricsId = this.preferencesStore.getState().metaMetricsId
         if (metametricsId && txMeta.swapMetaData && txReceipt.status !== '0x0') {
+          const tokensReceived = getSwapsTokensReceivedFromTxMeta(
+            txMeta.destinationTokenSymbol,
+            txMeta,
+            txMeta.destinationTokenAddress,
+            txMeta.txParams.from,
+            txMeta.destinationTokenDecimals,
+          )
+          const quoteVsExecutionRatio = `${(new BigNumber(tokensReceived, 10))
+            .div(txMeta.swapMetaData?.['token_to_amount'], 10).times(100).round(2)}%`
```

Comment:
Is `txMeta.swapMetaData` guaranteed to exist? This will thrown if `txMeta.swapMetaData.['token_to_amount']` isn't parsable by BigNumber as a number. So either that `?` isn't required, or we need some sort of fallback behaviour here.

Reply:
Oh, I think this is OK. I see that the condition for this block checks that `txMeta.swapMetaData` is truthy, and that the only place it's set does guarantee that `token_to_amount` is set.